Login MFA docs (#14317)
hashicorp/vault: website/content/api-docs/system/mfa/validate.mdx

---
layout: api
page_title: /sys/mfa/validate - HTTP API
description: >-
  The '/sys/mfa/validate' endpoint focuses on validating login MFA requests.
  If validation succeeds, it returns an auth response which includes a client token.
---

## Validate Login MFA Request

This endpoint validates a login request which is subject to MFA validation.

| Method | Path                |
| :----- | :------------------ |
| `POST`  | `/sys/mfa/validate` |

### Parameters

- `mfa_request_id` `(string: <required>)` – A unique identification of an MFA restricted login request.
 This can be found in the MFA requirement included in the auth response of the login request.

- `mfa_payload` `(map: <required>)` - A map of login MFA methodIDs to passcode credentials.

>  In cases where an MFA method is configured not to use passcodes, the passcode remains an empty string.

Author: Yeah, that is a good point. I tried to explain a bit in the login MFA procedure. But, it would be worth having similar things here as well.

This endpoint validates a login request which is subject to MFA validation.

| Method | Path                |
| :----- | :------------------ |
| `POST`  | `/sys/mfa/validate` |

### Parameters

- `mfa_request_id` `(string: <required>)` – A unique identification of an MFA restricted login request.
 This can be found in the MFA requirement included in the auth response of the login request.

- `mfa_payload` `(map<string|[]string>: <required>)` - A map of login MFA methodIDs to passcode credentials.
MFA methodIDs are UUID strings which are used as keys of the map. The values of the map are
string slices. In cases where an MFA method is configured not to use passcodes, the passcode remains an empty string.

### Sample Payload

```json
{
  "mfa_request_id": "5879c74a-1418-1948-7be9-97b209d693a7",
  "mfa_payload": {
      "d16fd3c2-50de-0b9b-eed3-0301dadeca10": ["910201"]
  }
}
```

### Sample Request

```shell-session
$ curl \
    --header "X-Vault-Token: ..." \
    --request POST \
    --data @payload.json \
    http://127.0.0.1:8200/v1/sys/mfa/validate
```

### Sample Response

In cases where MFA validation fails, a 403 status code is returned with
the details about the error.

> If MFA validation succeeds, the response will be the same as a successful

Reviewer: ```suggestion
If MFA validation succeeds, the response is identical to a successful
```
